### PR TITLE
FormDisplayTemplate function displayInput upgrading with twig templates

### PR DESCRIPTION
**Pull request by tabraiz961**
### Description

Please describe your pull request.

Fixes #

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

**Comment by tabraiz961** [comment]
signoff remains will push with the next commit.

File: libraries/classes/Config/FormDisplayTemplate.php [hunk 1]
```diff
@@ -177,18 +177,18 @@ public function displayInput(
         $opts = null
     ): string {
         static $icons;    // An array of IMG tags used further below in the function
-
+        $pathOptSpecialChars = '';
+        $setValueOptSpecialChars = '';
+        $check = '';
+        $pathSpecialChars = htmlspecialchars($path);
```
Comment (mauriciofauth) [inline libraries/classes/Config/FormDisplayTemplate.php]: Twig automatically escapes special chars, so you don't need to escape `$path` here.

**Review by MauricioFauth — CHANGES_REQUESTED** [review]

**Comment by williamdes** [comment]
@tabraiz961 you seem to have to many or strange spaces in your signoff line :)

**Comment by tabraiz961** [comment]
> @tabraiz961 you seem to have to many or strange spaces in your signoff line :)

All i had to do was increase the width of sidebar in GitKraken

**Comment by tabraiz961** [comment]
Hi @williamdes 
random order test failed. Is this something that i should be worry about.

**Comment by williamdes** [comment]
> Hi @williamdes
random order test failed. Is this something that i should be worry about.

Hi @tabraiz961 
No, they are failing on master branch before your PR. You can see that they are allowed to fail in `travis-ci.yml`

**Review by williamdes — APPROVED** [review]
Except the htmlspecialchars LGTM

File: libraries/classes/Config/FormDisplayTemplate.php [hunk 2]
```diff
@@ -170,18 +170,17 @@ public function displayInput(
         $opts = null
     ): string {
         static $icons;    // An array of IMG tags used further below in the function
-
+        $pathOpt = '';
+        $setValueOpt = '';
+        $check = '';
+        $nameSpecialCharsDecode = htmlspecialchars_decode($name);
```
Comment (williamdes) [inline libraries/classes/Config/FormDisplayTemplate.php]: Please remove this one :)

**Review by williamdes — CHANGES_REQUESTED** [review]

File: libraries/classes/Config/FormDisplayTemplate.php [hunk 3]
```diff
@@ -330,68 +301,52 @@ public function displayInput(
...
-            $htmlOutput .= '</td>';
-        } elseif ($isSetupScript) {
-            $htmlOutput .= '<td>&nbsp;</td>';
         }
-        $htmlOutput .= '</tr>';
-        return $htmlOutput;
+        return $this->template->render('/config/form_display/inputs', [
+            'tr_class' => $trClass,
+            'path' => $path,
+            'name_special_chars_decode' => $name,
```
Comment (williamdes) [inline libraries/classes/Config/FormDisplayTemplate.php]:
```suggestion
            'name' => $name,
```

**Review by williamdes — CHANGES_REQUESTED** [review]

File: libraries/classes/Config/FormDisplayTemplate.php [hunk 4]
```diff
@@ -229,8 +227,8 @@ public function displayInput(
         $hasErrors = isset($opts['errors']) && ! empty($opts['errors']);
         $optionIsDisabled = ! $isSetupScript && isset($opts['userprefs_allow'])
             && ! $opts['userprefs_allow'];
-        $nameId = 'name="' . htmlspecialchars($path) . '" id="'
-            . htmlspecialchars($path) . '"';
+        $nameId = 'name="' . $path . '" id="'
```
Comment (williamdes) [inline libraries/classes/Config/FormDisplayTemplate.php]: this change might introduce a security issue, please revert it

**Review by williamdes — CHANGES_REQUESTED** [review]
LGTM except the function removal :)

**Review by williamdes — APPROVED** [review]

File: templates/config/form_display/inputs.twig [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        {% if opts.show_restore_default is not empty %}
+            <a class="restore-default hide" href="#{{ path }}" title="{{ 'Restore default value'|trans }}">{{ icons.reload|raw }}</a>
+        {% endif %}
+        {% if has_errors == true %}
+            <dl class="inline_errors">{% for error in opts.errors %}<dd>{{ error|e('html') }}</dd>{% endfor %}</dl>
+        {% endif %}
+    </td>
+    {% if is_setup_script == true and opts.userprefs_allow is not empty %}
+        <td class="userprefs-allow" title="{{ 'Allow users to customize this value'|trans }}">
+            <input type="checkbox" name="{{ path }}-userprefs-allow" {{ check|raw }}
```
Comment (mauriciofauth) [inline templates/config/form_display/inputs.twig]:
```suggestion
            <input type="checkbox" name="{{ path }}-userprefs-allow"{{ opts.userprefs_allow ? ' checked' }}>
```

**Review by MauricioFauth — CHANGES_REQUESTED** [review]

**Comment by tabraiz961** [comment]
Hi @williamdes  i have moved this task to a seperate branch. According to GIT guide.
Is there a way where you can either duplicate or drop these commits? Thanks

**Comment by williamdes** [comment]
> Hi @williamdes i have moved this task to a seperate branch. According to GIT guide.
> Is there a way where you can either duplicate or drop these commits? Thanks

Hi @tabraiz961 
Can you post a link to the other PR and close this one ?

**Pull request closed without merging**